src/schema/rules/sidecars/mrs.yaml

```
@@ -0,0 +1,89 @@
+#
+# Groups of related metadata fields
+#
+# Assumptions: never need disjunction of selectors
+# Assumptions: top-to-bottom overrides is sufficient logic
+
+---
+# MRS Common metadata fields
+MRSScannerHardware:
+  selectors:
+    - modality == "mrs"
+  fields:
+    InstitutionName: recommended
+    InstitutionAddress: recommended
+    InstitutionalDepartmentName: recommended
+    Manufacturer: recommended
+    ManufacturersModelName: recommended
+    DeviceSerialNumber: recommended
+    StationName: recommended
+    SoftwareVersions: recommended
+    HardcopyDeviceSoftwareVersion: DEPRECATED
+    MagneticFieldStrength: recommended
+    ReceiveCoilName: recommended
+    ReceiveCoilActiveElements: recommended
+    NumberReceiveCoilActiveElements: recommended
+
+MRSSequenceSpecifics:
+  selectors:
+    - modality == "mrs"
+  fields:
+    PulseSequenceType:
+      level: recommended
+      description_addendum: |
+        For example, `"sLASER"`, `"MEGA-PRESS"`, `"EPSI"`, `"Metabolite-cycled MRSI"`.
+    ScanningSequence__mrs: recommended
+    SequenceName: recommended
+    PulseSequenceDetails: recommended
+    MRAcquisitionType__mrs: recommended for MRSI
+
+MRSRequiredFields:
+  selectors:
+    - modality == "mrs"
+  fields:
+    ResonantNucleus: required
+    SpectrometerFrequency: required
+    SpectralWidth: required
+    EchoTime: required
+
+MRSRecommendedFields:
+  selectors:
+    - modality == "mrs"
+  fields:
+    NumberOfSpectralPoints: recommended
+    NumberOfTransients: recommended for SVS
```

Is this optional for non-SVS data, or does it only apply to SVS data?

If it's generally optional, but promoted to recommended for SVS, then we want to make it clear here, and then add a second table that will be used for validation:

```YAML
MRSRecommendedFields:
  selectors:
    - modality == "mrs"
  fields:
    NumberOfSpectralPoints: recommended
    NumberOfTransients:
      level: optional
      level_addendum: recommended for SVS
    ...

SVSRecommendedFields:
  selectors:
    - datatype == "mrs"
    - suffix == "svs"
  fields:
    NumberOfTransients: recommended
```

If it only makes sense for SVS, then we'll want to remove it from this table, and add the second table like so:

```YAML
SVSRecommendedFields:
  selectors:
    - datatype == "mrs"
    - suffix == "svs"
  fields:
    NumberOfTransients:
      level: recommended
      level_addendum: for SVS
```

In this second case, the macro will need to combine the two fields:

```
{{ MACROS___make_sidecar_table(["mrs.MRSRecommendedFields", "mrs.SVSRecommendedFields"]) }}
```

---

It only makes sense for SVS, so the second case is more appropriate. Looping in @martin3141 and @wtclarke to confirm.

---

Agree mostly with Mark, on his assessment. Though you could sometimes define it for mrsi, it isn't generally a parameter of interest for those types. However, for unlocalised spectroscopy (`unloc`) then again it would be of interest like svs.

---

Good point; it also applies to `*_unloc` data.

---

I think it would make sense to just call it `OPTIONAL but RECOMMENDED for SVS data`, which would correspond to option 1.

---

It also applies to `unloc` data, so it should be `RECOMMENDED` for such data. Can we add: `level_addendum: for SVS and unloc`? It shouldn't be `OPTIONAL`, given how important it is to report `NumberOfTransients` in MRS papers (https://doi.org/10.1002/nbm.4484).

---

The way we would do this is:

```suggestion
    NumberOfTransients:
      level: optional
      level_addendum: recommended for SVS and unloc
```

This will be used for rendering.

And then add a separate table for enforcing:

```YAML
NumberOfTransientsRecommended:
  selectors:
    - datatype == "mrs"
    - intersects([suffix], ["svs", "unloc"])
  fields:
    NumberOfTransients: recommended
```

---

This sounds like the way forward to me!

---

I've updated the file to reflect the last suggestion.